test/simulator/timestamp/src/connection_simulator.cpp

```diff
@@ -52,13 +55,111 @@ connection_simulator::query_timestamp() const
     return (0);
 }
 
+/*
+ * Parse a single timestamp configuration string eg. oldest_timestamp=10. The string is split on the
+ * '=' and the new timestamp is assigned.
+ */
+int
+connection_simulator::parse_timestamp_config_single(
+  std::string config, uint64_t *new_oldest_ts, uint64_t *new_stable_ts)
+{
+
+    std::string ts_type;
+    uint64_t ts;
+
+    /* The substring before the '=' indicates the type of timestamp to be set. */
+    ts_type = config.substr(0, config.find("="));
+    /* Copy the substring after the '=' to get the timestamp value from the config string. */
+    std::string ts_string = config.substr(config.find("=") + 1);
+    /* Convert the timestamp to an int. */
+    ts = std::stoi(ts_string);
+
+    switch (system_timestamps_map[ts_type]) {
+    case oldest_timestamp:
+        *new_oldest_ts = ts;
+        break;
+    case stable_timestamp:
+        *new_stable_ts = ts;
+        break;
+    case durable_timestamp:
+        durable_ts = ts;
+        break;
+    }
+
+    return 0;
+}
+
+/*
+ *  Parse the timestamp configuration string with timestamps separated by ','.
+ */
+int
+connection_simulator::parse_timestamp_config(
+  std::string config, uint64_t *new_oldest_ts, uint64_t *new_stable_ts)
+{
+    size_t pos = 0;
+    std::string token;
+
+    /* Loop over the timestamp configuration strings separated by ','. */
+    while ((pos = config.find(",")) != std::string::npos) {
+        token = config.substr(0, pos);
+        std::cout << token << std::endl;
+
+        parse_timestamp_config_single(token, new_oldest_ts, new_stable_ts);
+
+        config.erase(0, pos + 1);
+    }
+
+    /* Parse the final timestamp configuration string. */
+    parse_timestamp_config_single(config, new_oldest_ts, new_stable_ts);
+
+    return 0;
+}
+
 int
-connection_simulator::set_timestamp() const
+connection_simulator::set_timestamp(std::string config)
 {
+    /* Set the new stable and oldest timestamps to the previous global values by default. */
+    uint64_t new_stable_ts = stable_ts;
+    uint64_t new_oldest_ts = oldest_ts;
+
+    parse_timestamp_config(config, &new_oldest_ts, &new_stable_ts);
+
+    /* Validate the new oldest timestamp if it is being updated. */
+    if (new_oldest_ts != oldest_ts &&
+      ts_mgr->validate_oldest_ts(new_stable_ts, new_oldest_ts) != 0) {
+        return 1;
+    }
+
+    /* Validate the new stable timestamp if it is being updated. */
+    if (new_stable_ts != stable_ts &&
+      ts_mgr->validate_stable_ts(new_stable_ts, new_oldest_ts) != 0) {
+        return 1;
+    }
+
+    /*
+     * The new timestamps have been validated up to this point. Now we can update the connection
+     * timestamps.
+     */
+    stable_ts = new_stable_ts;
+    oldest_ts = new_oldest_ts;
+
     return (0);
 }
 
-connection_simulator::connection_simulator() {}
+void
+connection_simulator::system_timestamps_map_setup()
+{
+    system_timestamps_map["oldest_timestamp"] = oldest_timestamp;
```

Comment:
Are we overdoing enums?

Reply:
Yeah this is probably unecessary since there are only three timestamps. Should I replace them with if statements?

Reply:
yup that sounds good